specification/search/data-plane/Azure.Search/preview/2025-08-01-preview/searchservice.json

```diff
@@ -0,0 +1,14691 @@
+          "type": "boolean",
+          "description": "Include reference source data by default."
+        },
+        "defaultMaxDocsForReranker": {
+          "type": "integer",
+          "description": "Default maximum documents for reranker."
+        },
+        "sourceDataSelect": {
+          "type": "string",
+          "description": "Used to request additional fields for referenced source data."
+        }
+      },
+      "required": [ "searchIndexName" ],
+      "description": "Parameters for search index knowledge source."
+    },
+    "AzureBlobKnowledgeSource": {
+      "type": "object",
+      "x-ms-discriminator-value": "azureBlob",
+      "allOf": [
+        {
+          "$ref": "#/definitions/KnowledgeSource"
+        }
+      ],
+      "properties": {
+        "azureBlobParameters": {
+          "$ref": "#/definitions/AzureBlobKnowledgeSourceParameters",
+          "description": "The type of the knowledge source."
+        }
+      },
+      "required": [ "azureBlobParameters" ],
+      "description": "Configuration for Azure Blob Storage knowledge source."
+    },
+    "AzureBlobKnowledgeSourceParameters": {
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/SearchIndexerDataIdentity",
+          "x-nullable": true,
+          "description": "An explicit managed identity to use for this datasource. If not specified and the connection string is a managed identity, the system-assigned managed identity is used. If not specified, the value remains unchanged. If \"none\" is specified, the value of this property is cleared."
+        },
+        "connectionString": {
+          "type": "string",
+          "description": "Key-based connection string or the ResourceId format if using a managed identity."
+        },
+        "containerName": {
+          "type": "string",
+          "description": "The name of the blob storage container."
+        },
+        "folderPath": {
+          "type": "string",
+          "description": "Optional folder path within the container."
+        },
+        "embeddingModel": {
+          "$ref": "#/definitions/VectorSearchVectorizer",
+          "description": "Optional vectorizer configuration for vectorizing content.",
+          "x-nullable": true
+        },
+        "chatCompletionModel": {
+          "$ref": "#/definitions/KnowledgeAgentModel",
+          "description": "Optional chat completion model for image verbalization or context extraction.",
+          "x-nullable": true
+        },
+        "ingestionSchedule": {
+          "$ref": "#/definitions/IndexingSchedule",
+          "description": "Optional schedule for data ingestion.",
+          "x-nullable": true
+        },
+        "createdResources": {
+          "$ref": "#/definitions/CreatedResources",
+          "description": "Resources created by the knowledge source.",
+          "readOnly": true
+        }
+      },
+      "required": [ "connectionString", "containerName" ], 
```

Comment:
Clarity question here, can `connectionString` be left empty if an `identity` is supplied? If so, should `connectionString` be removed as being required.